From 8d7d2f63e162551ec745d9081aa8423b5c5b5e86 Mon Sep 17 00:00:00 2001
From: Yiwei Zhang <zzyiwei@google.com>
Date: Tue, 21 Apr 2020 17:54:15 -0700
Subject: [PATCH 690/830] ANDROID: init: GKI: enable hidden configs for GPU
X-NVConfidentiality: public

Add hidden configs to GKI_HACKS_TO_FIX so they are enabled for loadable
GPU modules built out-of-tree.

Bug: 154525079
Test: rebuild kernel binary and pass checkvintf
Change-Id: I51871132b6a0bd1a55f5db7a9f90177cbc20ef86
Signed-off-by: Yiwei Zhang <zzyiwei@google.com>
---
 init/Kconfig.gki | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/init/Kconfig.gki b/init/Kconfig.gki
index b342a9434b7f..399e6c7a3ba5 100644
--- a/init/Kconfig.gki
+++ b/init/Kconfig.gki
@@ -149,6 +149,14 @@ config GKI_HIDDEN_RPMSG_CONFIGS
 	  If you are not building a kernel to be used for a variety of
 	  out-of-kernel build RPMSG drivers, say N here.
 
+config GKI_HIDDEN_GPU_CONFIGS
+	bool "Hidden GPU configuration needed for GKI"
+	select TRACE_GPU_MEM
+	help
+	  Dummy config option used to enable the hidden GPU config.
+	  These are normally selected implicitly when a module
+	  that relies on it is configured.
+
 # Atrocities needed for
 # a) building GKI modules in separate tree, or
 # b) building drivers that are not modularizable
@@ -172,6 +180,7 @@ config GKI_HACKS_TO_FIX
 	select GKI_HIDDEN_USB_CONFIGS
 	select GKI_HIDDEN_SOC_BUS_CONFIGS
 	select GKI_HIDDEN_RPMSG_CONFIGS
+	select GKI_HIDDEN_GPU_CONFIGS
 	help
 	  Dummy config option used to enable core functionality used by
 	  modules that may not be selectable in this config.
-- 
2.17.1

